### PR TITLE
test: add the unittest of module_base/matrix.h

### PR DESCRIPTION
**Pull request by pxlxingliang**
range: source/module_base/test/matrix_test.cpp

### CODE REVIEW

File: source/module_base/test/matrix_test.cpp [hunk 1]
```diff
@@ -0,0 +1,334 @@
...
+        for (int i=1;i<=9;++i) {m33b.c[i-1] = i*11.1;}
+
+        m33c.create(3,3,true);
+        m34a.create(3,4,true);
+        m34b.create(3,4,true);
+    }
+
+};
+
+TEST(matrix,Constructornrnc)
```
Comment (caic99) [inline source/module_base/test/matrix_test.cpp]: Please follow upper camel case naming.

File: source/module_base/test/matrix_test.cpp [hunk 2]
```diff
@@ -0,0 +1,334 @@
...
+{
+    ModuleBase::matrix m(3,4,true);
+    EXPECT_EQ(m.nr,3);
+    EXPECT_EQ(m.nc,4);
+    EXPECT_DOUBLE_EQ(m(0,0),0.0);
+    //EXPECT_DEATH(ModuleBase::matrix m(0,1),"");
```
Comment (caic99) [inline source/module_base/test/matrix_test.cpp]: Why commented out? What is the expected behavior if calling with a non-positive integer?

**Review by caic99 — CHANGES_REQUESTED** [review]

File: source/module_base/test/matrix_test.cpp (hunk 2 above)
Comment (caic99) [reply source/module_base/test/matrix_test.cpp:68]: > Why commented out? What is the expected behavior if calling with a non-positive integer?

Comment (pxlxingliang) [reply source/module_base/test/matrix_test.cpp]: Previous, I think the program should be terminated when construct a matrix by using a non-positive integer.  But, now, i think no one would use it as this case. So I will delete it.

Comment (caic99) [reply source/module_base/test/matrix_test.cpp]: > 



> Previous, I think the program should be terminated when construct a matrix by using a non-positive integer. But, now, i think no one would use it as this case. So I will delete it.

Agree. In practice, checking array access out of bound everytime results large overhead.
However, there are some differece between it and C++ convention. C++ array starts with 0, and this feature ensures 
 if you pass an index of type `size_t`(unsigned), [0] will access the first element.

**Review by caic99 — APPROVED** [review]

**Pull request merged**